### PR TITLE
Rewrite templates for launch

### PR DESCRIPTION
**Pull request by Omerr**
Overall - 10 templates.

### CODE REVIEW

File: .swm/doc-templates/a-flow-in-the-code-pqkt5.sw.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+## Introduction
+This doc describes the {{SUBJECT}} flow in our system. We will follow its implementation across the various locations so you can understand how the different parts create the full picture.
+
+## Following the flow
+
+<br/>
+<!-- TEMPLATE-swimm-snippet-placeholder -->
+Select the first step where the flow begins - e.g., the entry point, or an API call.
```
Comment (nadav-swimm) [inline .swm/doc-templates/a-flow-in-the-code-pqkt5.sw.md:15]: Shouldn't this also be {{Select the first step where the flow begins - e.g., the entry point, or an API call.}}

File: .swm/doc-templates/component-or-service-overview-pqkt4.sw.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Main features
+The main features of {{COMPONENT NAME}} are:
+
+## Interface
+{{How can this component/service be accessed?}}
+
+<br/>
+<!-- TEMPLATE-swimm-snippet-placeholder -->
+Select an example of accessing this component/service
```
Comment (nadav-swimm) [inline .swm/doc-templates/component-or-service-overview-pqkt4.sw.md:19]: Instructions. should be {{Select an example of accessing this component/service}}

File: .swm/doc-templates/engineering-design-pqkt8.sw md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+* Other relevant documents
+* Link to PRD
+  
+  *
+## Goals
+
+Main KPIs of this feature
+
+## High level design
+* Flow chart
```
Comment (nadav-swimm) [inline .swm/doc-templates/engineering-design-pqkt8.sw md]: {{Flow chart - Use "/mermaid" to add a flow chart}}

File: .swm/doc-templates/how-to-add-something.92m45.t.sw.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Some examples of {{SOMETHING}}s are {{examples}}.`
+
+## Basic example
+
+We'll follow the implementation of {{EXAMPLE (an instance of the SOMETHING, e.g., a sub-class)}} for this example.
+
+Every time we add a new instance, we follow these steps:
+<br/>
+<!-- TEMPLATE-swimm-snippet-placeholder -->
+Add a snippet of the first step needed to implement the EXAMPLE
```
Comment (nadav-swimm) [inline .swm/doc-templates/how-to-add-something.92m45.t.sw.md:23]: Instructions. {{Add a snippet of the first step needed to implement the EXAMPLE}}

File: .swm/doc-templates/a-flow-in-the-code-pqkt5.sw.md (hunk 1 above)
Comment (Omerr) [reply .swm/doc-templates/a-flow-in-the-code-pqkt5.sw.md:15]: @nadav-swimm this is within the snippet placeholder - so it appears within the placeholder.. I think we shouldn't use {{ for these...

File: .swm/doc-templates/component-or-service-overview-pqkt4.sw.md (hunk 2 above)
Comment (Omerr) [reply .swm/doc-templates/component-or-service-overview-pqkt4.sw.md:19]: @nadav-swimm this is within the snippet placeholder - so it appears within the placeholder.. I think we shouldn't use {{ for these...

File: .swm/doc-templates/how-to-add-something.92m45.t.sw.md (hunk 4 above)
Comment (Omerr) [reply .swm/doc-templates/how-to-add-something.92m45.t.sw.md:23]: @nadav-swimm this is within the snippet placeholder - so it appears within the placeholder.. I think we shouldn't use {{ for these...

File: .swm/doc-templates/component-or-service-overview-pqkt4.sw.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+<br/>
+
+## Main contributors
+{{Mention the main contributors to this module}}
```
Comment (saar-swimm) [inline .swm/doc-templates/component-or-service-overview-pqkt4.sw.md]: How about directory structure? Add more sections here, this seems a bit empty...
Use ideas from Content Suggestions (see here https://github.com/swimmio/swimm/blob/develop/packages/shared/src/content-suggestions/consts.ts)

Comment (saar-swimm) [reply .swm/doc-templates/component-or-service-overview-pqkt4.sw.md]: Regarding the 'main contributors', not sure how useful it really is here... at the very least maybe mention @mentions

File: .swm/doc-templates/engineering-design-pqkt8.sw md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+## Migrations
+
+## Security implications
+
+## Roll-out plan
+
+## KPIs
+
+What will be considered a success?
```
Comment (saar-swimm) [inline .swm/doc-templates/engineering-design-pqkt8.sw md]: Duplicate KPI

File: .swm/doc-templates/internal-api-pqkt6.sw.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+<br/>
+
+## Simple usage
+<br/>
+<!-- TEMPLATE-swimm-snippet-placeholder -->
+Show a simple example of using this API
+
+<br/>
+
+## Best practices and additional notes
```
Comment (saar-swimm) [inline .swm/doc-templates/internal-api-pqkt6.sw.md:35]: Add a section 'Advanced usage: {{explain scenario where this is needed}}'

File: .swm/doc-templates/product-requirements-document-pqkt7.sw md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+---
+id: pqkt7
+name: Product Requirements Document (PRD)
+file_version: 1.0.2
+app_version: 0.7.1-1
+---
+
+## Problem Statement
```
Comment (saar-swimm) [inline .swm/doc-templates/product-requirements-document-pqkt7.sw md]:
```suggestion
## Problem statement
```

**Review by saar-swimm — CHANGES_REQUESTED** [review]

File: .swm/doc-templates/component-or-service-overview-pqkt4.sw.md (hunk 5 above)
Comment (Omerr) [reply .swm/doc-templates/component-or-service-overview-pqkt4.sw.md]: I extended it a bit, but I am also refraining from overwhelming the user with too many sections

**Review by saar-swimm — APPROVED** [review]

**Pull request merged**